test/testshared/analysis_test.go

```diff
@@ -0,0 +1,138 @@
+package testshared
```

Comment:
Really nice tests to ensure parsing is working and document expected behavior! 🥇